### PR TITLE
Update persist auth snapshot

### PR DESCRIPTION
**Pull request by halink0803**

### CODE REVIEW

**Comment by halink0803** [comment]
[Return error if get blockchain balance error.](https://trello.com/c/O6AcVMni/151-return-error-if-get-blockchain-balance-error)

File: data/fetcher/fetcher.go [hunk 1]
```diff
@@ -529,6 +529,18 @@ func (self *Fetcher) PersistSnapshot(
 	}
 	// note: only update status when it's pending status
 	snapshot.ExchangeBalances = allEBalances
+
+	// persist blockchain balance
+	// if blockchain balance is not valid then auth snapshot will also not valid
+	for _, balance := range bbalances {
+		if !balance.Valid {
+			snapshot.Valid = false
+			if balance.Error != "" {
```
Comment (favadi) [inline data/fetcher/fetcher.go]: so, what happens if balance.Error is not empty? the error is ignored?

File: data/fetcher/fetcher.go [hunk 2]
```diff
@@ -529,6 +529,18 @@ func (self *Fetcher) PersistSnapshot(
...
+				snapshot.Error += " " + balance.Error
```
Comment (favadi) [inline data/fetcher/fetcher.go]: Could you add a `;` between error messages?

File: data/fetcher/fetcher.go (hunk 1 above)
Comment (halink0803) [reply data/fetcher/fetcher.go]: I thought it would be non-sense to add an empty string. I added the error message.

**Pull request merged**